testing/trino-tests/src/test/java/io/trino/tests/TestLocalQueries.java

```diff
@@ -117,4 +119,14 @@ public void testTransformValuesInTry()
                         "FROM (VALUES map(ARRAY[1, 2], ARRAY[0, 0]),  map(ARRAY[28], ARRAY[2]), map(ARRAY[18], ARRAY[2]), map(ARRAY[4, 5], ARRAY[1, 0]),  map(ARRAY[12], ARRAY[3])) AS t(m)",
                 "VALUES NULL, '{\"28\":14}', '{\"18\":9}', NULL, '{\"12\":4}'");
     }
+
+    @Test
+    public void testExtremelyLargeIn()
+    {
+        // query should not fail
+        String longValues = range(0, 100000)
+                .mapToObj(Integer::toString)
+                .collect(joining(", "));
+        computeActual("SELECT orderkey FROM orders WHERE orderkey IS NOT NULL OR orderkey IN (" + longValues + ")");
```

Comment:
`orderkey IS NOT` should probably by on other column to avoid interacting with `IN` clause